src/platform/user/profile/vap-svc/components/EmailField/EmailField.jsx

```diff
@@ -54,6 +54,7 @@ export default class EmailField extends React.Component {
         EditModal={EmailEditModal}
         formSchema={formSchema}
         uiSchema={uiSchema}
+        deleteDisabled={this.props.deleteDisabled}
```

Comment:
is this a supported prop on VAPServiceProfileField?

Reply:
Honestly, I copied the prop I saw in the [`AddressField`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/user/profile/vap-svc/components/AddressField/AddressField.jsx#L104) since it wasn't showing the delete button. The whole thing is pretty convoluted:

I don't see how passing `deleteDisabled` into `VAPServiceProfileField` does anything, but it works... somehow this prop is passed down to the [`VAPServiceEditModal`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/user/profile/vap-svc/components/base/VAPServiceEditModal.jsx#L91) which is then passed to [`VAPServiceEditModalActionButtons`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/user/profile/vap-svc/components/base/VAPServiceEditModalActionButtons.jsx#L65) as a `deleteEnabled` prop which then renders the delete button if `true`.

Reply:
Oh I see it now. Line 231 of the `VAPServiceProfileField` passes it to the `EmailEditModal` which in turn passes it to the `VAPServiceEditModal`. I really do not like blindly spreading props from component to component like this. This is why I want these components to just die and never be used again :(

Reply:
Oh? Should I (we) instead work on replacing these components? I'm not a fan of how this was all set up and the compromises I had to make to get it to work. You said that eventually we can update the profile data inline?

Reply:
If you want to add the features you are adding any time soon, you have to use these existing components. They work. I just hate the code. An inline editing solution is nowhere near being ready.